config/checkstyle/checkstyle.xml

```diff
@@ -225,19 +225,13 @@
     <module name="JavadocMethod">
       <property name="id" value="JavadocMethodMain"/>
       <property name="scope" value="public" />
-      <property name="allowUndeclaredRTE" value="true" />
-      <property name="allowThrowsTagsForSubclasses" value="false" />
       <property name="allowMissingParamTags" value="true" />
-      <property name="allowMissingThrowsTags" value="true" />
```

Comment:
I feel like these three removals will change what checkstyle allows through. Is there a replacement for them, or did we decide that it's not a big deal? Can you provide more details here?

Reply:
Their removal is non-backwards compatible, which means we have to remove them if we want to upgrade (or else checkstyle won't work): https://github.com/checkstyle/checkstyle/issues/7329
It seems they were removed because they were not reliable.

The suggested alternative from the issue I linked, would be to set `validateThrows` to `true` (by default, it is false), but doing that suddenly flags hundreds of javadoc comments that do not declare a `@throws` element, something we had not set to true before. And the fact that we had set `allowMissingThrowsTags` to `true` tells me we did not care about declaring `@throws` tags in method javadoc anyway, so I assumed we still don't.

If you think we should check every method that throws an exception must declare it in the javadoc (at least in main, not test), then I can get those changes in here.

Reply:
Nope, that's fine by me. Just wanted to make sure we weren't about to make a change that softened our requirements by accident.